Form/EventListener/ResizePolyFormListener.php

```diff
@@ -84,7 +85,12 @@ public function __construct(array $prototypes, array $options = array(), $allowA
                 $defaultType = $type;
             }
 
-            $typeKey = $type instanceof FormTypeInterface ? $type->getName() : $type;
+            $typeKey = $type;
+
+            if ($type instanceof FormTypeInterface) {
```

Comment:
@merk this is about accepting type instance in your own component. My point is that you should deprecate passing type instances in a poly-collection, to be consistent with Symfony which deprecated passing type instance anywhere in core types